care/abdm/api/viewsets/healthid.py

```diff
@@ -0,0 +1,655 @@
+
+        if ratelimit(request, "verify_mobile_otp", [data["txnId"]]):
+            raise CaptchaRequiredException(
+                detail={"status": 429, "detail": "Too Many Requests Provide Captcha"},
+                code=status.HTTP_429_TOO_MANY_REQUESTS,
+            )
+
+        serializer = VerifyOtpRequestPayloadSerializer(data=data)
+        serializer.is_valid(raise_exception=True)
+        response = HealthIdGateway().verify_mobile_otp(data)
+        return Response(response, status=status.HTTP_200_OK)
+
+    def create_abha(self, abha_profile, token):
+        abha_object = AbhaNumber.objects.filter(
+            abha_number=abha_profile["healthIdNumber"]
+        ).first()
+
+        if abha_object:
+            return abha_object
+
+        abha_object = AbhaNumber.objects.create(
+            abha_number=abha_profile["healthIdNumber"],
+            health_id=abha_profile["healthId"],
+            name=abha_profile["name"],
+            first_name=abha_profile["firstName"],
+            middle_name=abha_profile["middleName"],
+            last_name=abha_profile["lastName"],
+            gender=abha_profile["gender"],
+            date_of_birth=str(
+                datetime.strptime(
+                    f"{abha_profile['yearOfBirth']}-{abha_profile['monthOfBirth']}-{abha_profile['dayOfBirth']}",
+                    "%Y-%m-%d",
+                )
+            )[0:10],
+            address=abha_profile["address"] if "address" in abha_profile else "",
+            district=abha_profile["districtName"],
+            state=abha_profile["stateName"],
+            pincode=abha_profile["pincode"],
+            email=abha_profile["email"],
+            profile_photo=abha_profile["profilePhoto"],
+            txn_id=token["txn_id"],
+            access_token=token["access_token"],
+            refresh_token=token["refresh_token"],
+        )
+        abha_object.save()
+
+        return abha_object
+
+    def add_abha_details_to_patient(self, abha_object, patient_object):
+        # patient = PatientRegistration.objects.filter(
+        #     abha_number__abha_number=abha_object.abha_number
+        # ).first()
+
+        # if patient or patient_object.abha_number:
+        #     return False
+
+        patient_object.abha_number = abha_object
+        patient_object.save()
+        return True
+
+    @extend_schema(
+        # /v1/registration/aadhaar/createHealthId
+        operation_id="create_health_id",
+        request=CreateHealthIdSerializer,
+        responses={"200": "{'txnId': 'string'}"},
+        tags=["ABDM HealthID"],
+    )
+    @action(detail=False, methods=["post"])
+    def create_health_id(self, request):
+        data = request.data
+
+        if ratelimit(request, "create_health_id", [data["txnId"]]):
+            raise CaptchaRequiredException(
+                detail={"status": 429, "detail": "Too Many Requests Provide Captcha"},
+                code=status.HTTP_429_TOO_MANY_REQUESTS,
+            )
```

Comment:
rate limit is being repeated, can we use the standard implementation

Reply:
no, this was specifically requested by wasa team